RecFCCeeCalorimeter/src/components/CreateFCCeeCaloXTalkNeighbours.cpp

```diff
@@ -0,0 +1,256 @@
+    return StatusCode::FAILURE;
+  }
+  std::unordered_map<uint64_t, std::vector<std::pair<uint64_t,double>>> map;
+#if DEBUGCELLINFO ==1
+  std::unordered_map<uint64_t, std::vector<int>> CellInfo;
+#endif
+
+  for (uint iSys = 0; iSys < m_readoutNamesSegmented.size(); iSys++)
+  {
+    // Check if readout exists
+    info() << "Readout: " << m_readoutNamesSegmented[iSys] << endmsg;
+    if (m_geoSvc->getDetector()->readouts().find(m_readoutNamesSegmented[iSys]) == m_geoSvc->getDetector()->readouts().end())
+    {
+      error() << "Readout <<" << m_readoutNamesSegmented[iSys] << ">> does not exist." << endmsg;
+      return StatusCode::FAILURE;
+    }
+
+    // get theta-based segmentation
+    dd4hep::DDSegmentation::Segmentation *aSegmentation = m_geoSvc->getDetector()->readout(m_readoutNamesSegmented[iSys]).segmentation().segmentation();
+    if (aSegmentation == nullptr)
+    {
+      error() << "Segmentation does not exist." << endmsg;
+      return StatusCode::FAILURE;
+    }
+
+    std::string segmentationType = aSegmentation->type();
+    info() << "Segmentation type : " << segmentationType << endmsg;
+
+    dd4hep::DDSegmentation::GridTheta_k4geo *segmentation = nullptr;
+    dd4hep::DDSegmentation::FCCSWGridModuleThetaMerged_k4geo *moduleThetaSegmentation = nullptr;
+    if (segmentationType == "FCCSWGridModuleThetaMerged_k4geo")
+    {
+      segmentation = dynamic_cast<dd4hep::DDSegmentation::GridTheta_k4geo *>(aSegmentation);
+      moduleThetaSegmentation = dynamic_cast<dd4hep::DDSegmentation::FCCSWGridModuleThetaMerged_k4geo *>(aSegmentation);
+    }
+    else
+    {
+      error() << "Segmentation type not handled." << endmsg;
+      return StatusCode::FAILURE;
+    }
+
+    if (segmentation == nullptr || moduleThetaSegmentation == nullptr)
+    {
+      error() << "Unable to cast segmentation pointer!!!!" << endmsg;
+      return StatusCode::FAILURE;
+    }
+
+    info() << "Segmentation: size in Theta " << segmentation->gridSizeTheta() << endmsg;
+    info() << "Segmentation: offset in Theta " << segmentation->offsetTheta() << endmsg;
+    if (segmentationType == "FCCSWGridModuleThetaMerged_k4geo")
+    {
+      info() << "Segmentation: bins in Module " << moduleThetaSegmentation->nModules() << endmsg;
+    }
+
+    // retrieve decoders for the crosstalk neighbour finding
+    auto decoder = m_geoSvc->getDetector()->readout(m_readoutNamesSegmented[iSys]).idSpec().decoder();
+
+    // Loop over all cells in the calorimeter and retrieve existing cellIDs and find neihbours
+    if (segmentationType == "FCCSWGridModuleThetaMerged_k4geo")
+    {
+      std::vector<std::vector<std::pair<int, int>>> extrema_layer;
+      for (unsigned int ilayer = 0; ilayer < m_activeVolumesNumbersSegmented[iSys]; ilayer++)
+      {
+        dd4hep::DDSegmentation::CellID volumeId = 0;
+        (*decoder)[m_fieldNamesSegmented[iSys]].set(volumeId, m_fieldValuesSegmented[iSys]);
```

Comment:
Can you use C++ casts (`static_cast<>()`)instead of C ones (`()`)?

Reply:
I have changed all "(*decoder)"'s in my code to "static_cast<dd4hep::DDSegmentation::BitFieldCoder>(*decoder)"